lib/python/rosie/svn_pre_commit.py

```diff
@@ -246,22 +261,29 @@ def run(self, repos, txn):
             if (self.ST_ADDED, path_head + "trunk/") in changes:
                 continue
 
-            # See whether author has permission to make changes
-            if author is None:
-                author = self._svnlook("author", "-t", txn, repos).strip()
-            if super_users is None:
+            if not found_author_info:
+                author = self._get_author(txn, repos)
+                author_aliases = self._get_author_aliases(repos)
                 super_users = []
                 for s_key in ["rosa-svn", "rosa-svn-pre-commit"]:
                     value = conf.get_value([s_key, "super-users"])
                     if value is not None:
                         super_users = shlex.split(value)
                         break
+                found_author_info = True
+
+            # See whether author has permission to make changes
             if sid not in rev_info_map:
                 rev_info_map[sid] = self._load_info(
                     repos, sid, branch=branch)
             owner, access_list = self._get_access_info(rev_info_map[sid])
             admin_users = super_users + [owner]
 
+            if author in author_aliases and author_aliases[author] == owner:
```

Comment:
```suggestion
            if author_aliases.get(author) == owner:
```
Any reason not do this?

Reply:
I was a bit worried about None owners... it was a bit irrational